### PR TITLE
Include helper methods for the LanguagePlatform.

### PR DESCRIPTION
**Pull request by Impertatore**
The SegmentPair processer offers the possiblity to recover additional information from the ISegmentPair (FileTypeSupport.Framework)
through the LanguagePlatform, that includes the tokenized segments and word counts.

The processor creates a tempoaray translation memory given the settings provided; the TM is then used to recover the LanguagePlatform
segment, which is a sequence of Sdl.LanguagePlatform.Core.SegmentElements.

### CODE REVIEW

File: Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		// Summary:
+		//     The default product path, created in the my documents folder
+		public virtual string ProductPath
+		{
+			get
+			{
+				if (!string.IsNullOrEmpty(_productPath))
+					return _productPath;
+
+				_productPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), _productName);
```
Comment (cromica) [inline Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs]: Shouldn't we use the Studio folder path? The toolkit is not a product. The toolkit core has a bunch of libraries that provide the information required for identifying current version of Studio.

**Review by cromica — CHANGES_REQUESTED** [review]

File: Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs (hunk 1 above)
Comment (Impertatore) [reply Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs]: The path information where the temporary resources are created and managed to provide this functionality is the responsibility of the client that is integrating with the functionality offered by the toolkit, i.e. thier product. The class called PathInfo includes default paths (and names), however this is the developers choice.

Wouldn’t be more correct to allow the developer to decide where these temporary resources are created & managed, with the expected defaults?

Comment (Impertatore) [reply Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs]: Just thinking this through again and might make more sense to manage the temporary resources used in providing the functionality in the Studio folder path as you suggested.  I'll update the PR later on this evening

Comment (cromica) [reply Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs]: Thanks Patrick. I agree that the developer should be able to decide about the path so having the PathInfo virtual is great. It's just around the default value.

**Pull request merged**